### ISSUE TITLE
Parse audio stream's channel count, language and index

### ISSUE THREAD

**Issue by alexeyabel** [nathanbabcock/ffmpeg-sidecar#49]
Hi Nathan,

AFAICS `ParsedInputStream`  events for `stream_type: "Audio"` don't extract the channel count, language or index but it is provided by ffmpeg:

```
ParsedInputStream: AVStream { stream_type: "Audio", format: "unknown", pix_fmt: "unknown", width: 0, height: 0, fps: 0.0, parent_index: 0, raw_log_message: "[info]   Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)" }
```

It would be nice to have this available as structured data. Maybe seperate `ParsedInputStream` into `ParsedAudioInputStream` and `ParsedVideoInputStream` etc? What do you think?

**Comment by nathanbabcock** [comment]
There is an entrypoint for it inside of `try_parse_stream` in `log_parser.rs`, but no logic implemented yet!

Totally agree with your idea of splitting into `ParsedAudioInputStream` and `ParsedVideoInputStream`. And then you might also need to split the inner struct `AVStream` into two separate ones, `VideoStream` and `AudioStream`, since the current one is pretty much all properties that only apply to video.

**Comment by nathanbabcock** [comment]
1. `input_streams` is not specifically required for any library functionality, it's just passed through as metadata that might be useful for an application that uses the library
2. `output_streams` on the other hand is used internally when piping from ffmpeg `stdout` directly into the iterator. It reads the output stream metadata (video only, like you said) in order to determine how many bytes to expect per frame — the main place it gets used is this little helper function: [`get_bytes_per_frame`](https://github.com/nathanbabcock/ffmpeg-sidecar/blob/ba459ef02fe343b21596ddd746f85521c4a44391/src/pix_fmt.rs#L230). Then it consumes that many bytes from stdout and groups them into `OutputFrame` events.

So TL;DR: as long as the output stream information about width, height, and pixel format gets preserved somewhere it should work fine.

Another thought I had was whether `AVStream` itself should be an enum, consisting of `VideoStream`, `AudioStream`, and `OtherStream` as you defined them. But I'm not really sure yet whether a nested enum (like `FfmpegEvent::ParsedInputStream(AVStream::Video(x))`) would be more or less intuitive than just sticking all the different variations into the top-level event enum.
